mmv1/templates/terraform/constants/tpu_vm.go.erb

```diff
@@ -29,3 +29,18 @@ func normalizeScopes(scopes []string) []string {
 	}
 	return result
 }
+
+// For backwards compatibility, we need to maintain original behavior where `accelerator_type`
+// defaults "v2-8" when nothing is set. However, if the newly introduced `accelerator_config` field
+// is set, then use that value instead of the `accelerator_type` default.
+func acceleratorTypeCustomizeDiff(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+	_, isTypeSet := diff.GetOk("accelerator_type")
+	_, isConfigSet := diff.GetOk("accelerator_config")
+	if !isTypeSet && !isConfigSet {
```

Comment:
Also check if either value is unknown- CDs get run multiple times, and the value may be unknown. We'll generate an inconsistent plan if we don't account for that and a user feeds in an unknown value.

I don't remember the call offhand, but it's in the docs for ResourceDiff.

Reply:
It looks like you mean [NewValueKnown](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/resource_diff.go#L419), and I'm a little unclear on the edge cases, but I think you're suggesting that we don't set this default value unless both values are "known" to be unset? I assume we will always eventually get a call where both values are known?

Reply:
Hmm, actually, maybe it's fine. That may just apply to errors

Reply:
Since in some cases they will never be known

Reply:
Hrm ok, maybe I'll need to do some testing then? It seems like assuming in either direction if a value is unknown would result in an invalid API call.